src/kernel/kernel.h

```diff
@@ -100,6 +97,18 @@ static inline void set_running_frame(context f)
     current_cpu()->running_frame = f;
 }
 
+static inline nanos_thread get_current_thread()
+{
+    context f = current_cpu()->kernel_context->frame;
+    return pointer_from_u64(f[FRAME_THREAD]);
```

Comment:
Just curious as to your reasoning for moving these back to the kernel context frame. I went back and forth between the two. It does seem that a kernel page fault on a user page in a syscall continuation could result in the continuation being resumed on another core...which would indeed advocate placing the thread in the frame.

Reply:
Yes, as I wrote in the commit message, a kernel context can be moved between cores, so the current thread should be stored in the frame of the kernel context.